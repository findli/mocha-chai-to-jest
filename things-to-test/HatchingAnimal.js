import React from 'react';

export default class HatchingAnimal extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            styleHeight: '1px',
        };
    }

    componentDidMount() {
        // N.B. This is one of the only case where you can do
        // a setState() inside a componentDidMount
        // Because it's ok for it to rerender
        this.setState({ // eslint-disable-line react/no-did-mount-set-state
            styleHeight: '150px',
        });
    }

    render() {
        const { styleHeight } = this.state;
        return (
            <div className="section" style={{ height: styleHeight }}>
                <p>
                    I am growing when I am mounted.
                </p>
            </div>
        );
    }
}
