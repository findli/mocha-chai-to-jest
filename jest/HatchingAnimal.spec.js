import React from 'react';
import ReactTestUtils from 'react-addons-test-utils';
import HatchingAnimal from '../things-to-test/HatchingAnimal';

describe('HatchingAnimal component', () => {
    it('renders correctly with min params', () => {
        const shallowRenderer = ReactTestUtils.createRenderer();
        const result = shallowRenderer.render(
            <HatchingAnimal />,
        );
        expect(result).toMatchSnapshot();
    });

    it('componentDidMount is working', () => {
        const shallowRenderer = ReactTestUtils.createRenderer();
        shallowRenderer.render(
            <HatchingAnimal />,
        );
        const instance = shallowRenderer.getMountedInstance();
        expect(instance.state.styleHeight).toBe('1px');
        instance.componentDidMount();
        expect(instance.state.styleHeight).toBe('150px');
    });
});
