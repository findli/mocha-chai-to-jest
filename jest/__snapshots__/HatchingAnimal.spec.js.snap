// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HatchingAnimal component renders correctly with min params 1`] = `
<div
  className="section"
  style={
    Object {
      "height": "1px",
    }
  }
>
  <p>
    I am growing when I am mounted.
  </p>
</div>
`;
